staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -79,29 +81,58 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 
 // CheckClient accepts client and calculate the support per endpoint and caches it.
 // It will return at any point if error happens or one endpoint is not supported.
-func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
+func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, cancel context.CancelFunc, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, cancel, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, cancel context.CancelFunc, c client) error {
+	go func(ctx context.Context) {
```

Comment:
I think that a new go routine will be created for each resource (`newStore`).
Maybe we should create a controller per etcd client instead ?

Reply:
That's a different perspective and i appreciate it, i'm not sure about changing, we should ask for @serathius and @wojtek-t opinion as well. 
But for reference can you provide me an example of what type of controller do you mean if there is any in the code? And where we should invoke controller?